api/v1alpha1/awsvalidator_types.go

```diff
@@ -40,6 +40,11 @@ type AwsAuth struct {
 	SecretName string `json:"secretName,omitempty" yaml:"secretName,omitempty"`
 	// Option 2: specify a service account (EKS)
 	ServiceAccountName string `json:"serviceAccountName,omitempty" yaml:"serviceAccountName,omitempty"`
+
+	// STS Options
+	RoleArn         string `json:"roleArn,omitempty" yaml:"roleArn,omitempty"`
```

Comment:
Please move the STS fields into a struct (`STSAuth` perhaps?) and add just that struct to the `AwsAuth` struct. That way the fields of the `STSAuth` struct can all be required (no `omitempty`) and the struct itself can be optional.

Also, notice how comments above each struct field are added to the CRD as a description for that field by swagger. So please either add an additional newline under `// STS Options` or change the comment to be a meaningful description for the field directly underneath it.

Reply:
Also, please include support for `ExternalID`